packages/protocol-perf/src/index.ts

```diff
@@ -177,15 +244,21 @@ class DefaultPerfService implements Startable, PerfService {
         throw new Error(`Expected to receive ${recvBytes} bytes, but received ${actualRecvdBytes}`)
       }
     } catch (err) {
-      log('error sending %i bytes to %p: %s', sendBytes, connection.remotePeer, err)
+      log('error sending %s bytes to %p: %s', totalBytesSent, connection.remotePeer, err)
       throw err
     } finally {
       log('performed %s to %p', this.protocol, connection.remotePeer)
       await stream.close()
     }
 
-    // Return the latency
-    return Date.now() - startTime
+    const finalOutput: PerfOutput = {
+      type: 'final',
+      timeSeconds: (Date.now() - initialStartTime) / 1000,
```

Comment:
`timeSeconds` should be calculated after the final byte has been read - if we calculate it here it includes the time taken to gracefully close the stream (e.g. the `await stream.close()`) in the `finally` block above.